src/cache.h

```diff
@@ -0,0 +1,42 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "list.h"
+
+/*
+    T1: LRU List
+    T2: LFU List
+    B1: LRU Ghost List
+    B2: LFU Ghost List
+*/
+typedef enum { T1, B1, T2, B2 } arc_type_t;
```

Comment:
Don't call it `arc_type_t`.
The implementation details should appear in C source rather than public headers.